app/ports/models/port.py

```diff
@@ -0,0 +1,335 @@
+                                except Port.DoesNotExist:
+                                    print("Failed to append {} as a dependency to {}. Not Found.".format(
+                                        depends.rsplit(':', 1)[-1],
+                                        port['name']))
+                            obj.save()
+                            obj.dependencies.add(*dependencies)
+
+                except Port.DoesNotExist:
+                    print(
+                        "Failed to update depencies for {}. Port does not exist in the database.".format(port['name']))
+
+        # Takes in a list of JSON objects and runs the updates
+        def run_updates(ports):
+            full_update_ports(ports)
+            full_update_dependencies(ports)
+
+        # Block to find type of passed "data" and run updates accordingly
+        # ============ START ============
+
+        # If a valid path is passed, open it and parse the JSON objects
+        if isinstance(data, str):
+            if os.path.exists(data):
+                ports = open_portindex_json(data)
+                run_updates(ports)
+            else:
+                print('File "{}" not found.'.format(data))
+
+        # If a list is passed, run the updates directly
+        elif isinstance(data, list):
+            run_updates(data)
+
+        # ============ END ==============
+
+    @classmethod
+    def mark_deleted(cls, dict_of_portdirs):
+        for portdir in dict_of_portdirs:
+            for port in Port.objects.filter(portdir__iexact=portdir).only('portdir', 'name', 'active'):
+                if port not in dict_of_portdirs[portdir]:
+                    port.active = False
+                    port.save()
+
+    class PortIndexUpdateHandler:
+        @staticmethod
+        def sync_and_open_file():
+            subprocess.call([config.RSYNC, config.PORTINDEX_SOURCE, config.PORTINDEX_JSON])
+            with open(config.PORTINDEX_JSON, "r", encoding='utf-8') as file:
+                data = json.load(file)
+            return data
+
+
+class Dependency(models.Model):
+    port_name = models.ForeignKey(Port, on_delete=models.CASCADE, related_name="dependent_port")
+    dependencies = models.ManyToManyField(Port)
+    type = models.CharField(max_length=100)
+
+    class Meta:
+        unique_together = [['port_name', 'type']]
+
+        indexes = [
+            models.Index(fields=['port_name'])
+        ]
+
+
+class Variant(models.Model):
+    port = models.ForeignKey(Port, on_delete=models.CASCADE, related_name='ports')
+    variant = models.CharField(max_length=100, default='')
+
+
+class Maintainer(models.Model):
+    name = models.CharField(max_length=50, default='')
+    domain = models.CharField(max_length=50, default='')
+    github = models.CharField(max_length=50, default='')
+    ports = models.ManyToManyField(Port, related_name='maintainers')
+
+    objects = PortManager()
```

Comment:
Is it okay to use the same manager for both the `Port` model as well as `Maintainer`? Will `name` be able to uniquely identify a maintainer? I'd go with separate ones in case we end up modifying manager class to suit needs for the `Port` model and overlooking `Maintainer` model fields.
I may have overlooked the use case around using a manager at all in our case.